File: self_dividing.ts
class selfDividing {
    private data:Array<number>

    constructor(d:Array<number>){
        this.data = d
    }
    public isConditionTrue():Array<number> {
        var f:Array<number> = []
        for (var x = 0; x < this.data.length; x++){
            if (this.isDividing(this.data[x])){
                f.push(this.data[x])
            }
        }
        return f;
    }
    private isDividing(f:number):boolean{
        var j = f.toString()
        for (var x = 0; x < j.length; x++){
            if (f % parseInt(j[x]) != 0){
                return false
            }
        }
        return true
    }
}

function selfDividingNumbers(left: number, right: number): number[] {
    var arr:Array<number> = new Array()
    for (var x = left; x < right + 1; x++){
        arr.push(x)
    }
    var v  = new selfDividing(arr);

    return v.isConditionTrue();
};

console.log(selfDividingNumbers(1, 22))

File: self_dividing.dart
class SelfDividing {
  List<int> arr;
  
  SelfDividing(List<int> arr){
    this.arr = arr;
  }
  
  List<int> isConditionTrue() {
    List<int> l;
    for(int x = 0; x < this.arr.length; x++){
      if(this.isDividing(this.arr[x])){
        l.add(this.arr[x]);
      }
    }
    return l;
  }
  
  bool isDividing(int data){
    var d = data.toString();
    for (int x = 0; x < d.length; x++){
      if(data % int.parse(d[x]) == 0){
        
      } else {
        return false;
      }
    }
    
    return true;
  }
}

void main() {
  List<int> arr = [];
  for(int x = 1; x < 22 + 1; x++){
    arr.add(x);
  }
  var data = new SelfDividing(arr);
  print(data);
}
